hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -975,7 +984,7 @@ func (r *NodePoolReconciler) reconcileMachineDeployment(log logr.Logger,
 		// Before persisting, if the NodePool is brand new we want to make sure the replica number is set so the machineDeployment controller
 		// does not panic.
 		if machineDeployment.Spec.Replicas == nil {
-			machineDeployment.Spec.Replicas = k8sutilspointer.Int32Ptr(k8sutilspointer.Int32PtrDerefOr(nodePool.Spec.Replicas, 0))
+			setMachineDeploymentReplicas(nodePool, machineDeployment)
```

Comment:
to clarify, is this change cosmetic or does it also intend to have an impact to reduce generation?

Reply:
It does have an impact in that for autoscaling nodepools the replicas would first be set to 0, then when this block was skipped they would get set to 1 by `setMachineDeploymentReplicas` resulting in another generation.